lib/kernel-manager.coffee

```diff
@@ -162,6 +164,23 @@ module.exports = KernelManager =
 
             onStarted?(kernel)
 
+        try
+            data = fs.readFileSync customKernelConnectionPath, 'utf8'
+            config = JSON.parse data
+            console.log "Using custom kernel connection: ", customKernelConnectionPath
+            kernel = new Kernel kernelSpec, grammar, config, customKernelConnectionPath, true
+            finishKernelStartup kernel
+        catch e
+            if e.code != 'ENOENT'
+                trow e
+            console.log(e)
+            ConfigManager.writeConfigFile (filepath, config, onlyConnect = false) =>
```

Comment:
This bit is confusing because `ConfigManager.writeConfigFile` expects a callback with two arguments only (`filepath` and `config`). I think as below would be clearer:

```coffee
            ConfigManager.writeConfigFile (filepath, config) =>
                kernel = new Kernel kernelSpec, grammar, config, filepath, onlyConnect=false
                finishKernelStartup kernel
```